python/core/auto_generated/fieldformatter/qgsvaluerelationfieldformatter.sip.in

```diff
@@ -121,6 +121,15 @@ Check whether the ``feature`` has all values required by the ``expression``
 .. versionadded:: 3.2
 %End
 
+    static QString resolveLayer( const QVariantMap &config );
+%Docstring
+Returns the (possibly empty) layer ID from the widget ``config``
```

Comment:
Actually, wouldn't it be nicer to directly return a weak layer pointer from here already?

Reply:
I chose this approach due to const-ness differences (in one case we need a not-const).

Reply:
The const-ness of the layer? Can't this just return a non-const layer which can then be passed and used as a const layer too?

Reply:
Of course: that was my first implementation, but the I thought that it may be breaking one of the many unwritten coding guidelines about constness.

Reply:
I don't think so.